docs/source/release_notes/release-notes.rst

```diff
@@ -55,6 +55,13 @@ develop
            There is no issue with having tables with different values for ``gc_grace_seconds``, and this can be updated at any time.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1726>`__)
 
+    *    - |fixed|
+         - A 1s timeout has been added to the client when the client has queried all the servers of a cluster and received a NotCurrentLeaderException.
```

Comment:
This release note is a bit wordy. It's probably enough to keep the first sentence and add a second: "This should reduce contention during leadership elections".

If we make the timeout configurable, we should add a link to the docs here.

Also, we should link to this PR.